e2e/utils/test_cases.js

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// See reference: https://support.smartbear.com/tm4j-cloud/api-docs/
+
+const axios = require('axios');
+
+const {getAllTests} = require('./report');
+
+const status = {
+    passed: 'Pass',
+    failed: 'Fail',
+    pending: 'Pending',
+    skipped: 'Skip',
+};
+
+const environment = {
+    chrome: 'Chrome',
+    firefox: 'Firefox',
+};
+
+function getStepStateResult(steps = []) {
+    return steps.reduce((acc, item) => {
+        if (acc[item.state]) {
+            acc[item.state] += 1;
+        } else {
+            acc[item.state] = 1;
+        }
+
+        return acc;
+    }, {});
+}
+
+function getStepStateSummary(steps = []) {
+    const result = getStepStateResult(steps);
+
+    let summary = '';
+    Object.entries(result).forEach(([key, value], index) => {
+        if (index) {
+            summary += ',';
+        }
+        summary += `${value} ${key}`;
+    });
+
+    return summary;
+}
+
+function getTM4JTestCases(report) {
+    const allTests = getAllTests(report.results);
+
+    const re = /(MM-T)\w+/g;
+    return allTests.
+        filter((item) => re.test(item.title)).
+        map((item) => {
+            return {
+                title: item.title,
+                duration: item.duration,
+                incrementalDuration: item.incrementalDuration,
+                state: item.state,
+                pass: item.pass,
+                fail: item.fail,
+                pending: item.pending,
+            };
+        }).
+        reduce((acc, item) => {
+            const key = item.title.split(' ')[0].split('_')[0];
```

Comment:
Quick question: I'm assuming these are the ticket numbers. Do they have underscore in them?

Reply:
Correct. I'll add in the description and to follow on E2E docs but for those test cases that have multiple test steps, the foreseen convention would be `[tm4j_key]_[step_number]`, e.g. `MM-T1_4`.